common/src/main/java/io/netty/util/internal/NativeLibraryLoader.java

```diff
@@ -257,25 +281,36 @@ public Object run() {
             throws ClassNotFoundException {
         try {
             return loader.loadClass(helper.getName());
-        } catch (ClassNotFoundException e) {
-            // The helper class is NOT found in target ClassLoader, we have to define the helper class.
-            final byte[] classBinary = classToByteArray(helper);
-            return AccessController.doPrivileged(new PrivilegedAction<Class<?>>() {
-                @Override
-                public Class<?> run() {
-                    try {
-                        // Define the helper class in the target ClassLoader,
-                        //  then we can call the helper to load the native library.
-                        Method defineClass = ClassLoader.class.getDeclaredMethod("defineClass", String.class,
-                                byte[].class, int.class, int.class);
-                        defineClass.setAccessible(true);
-                        return (Class<?>) defineClass.invoke(loader, helper.getName(), classBinary, 0,
-                                classBinary.length);
-                    } catch (Exception e) {
-                        throw new IllegalStateException("Define class failed!", e);
+        } catch (ClassNotFoundException e1) {
+            try {
+                // The helper class is NOT found in target ClassLoader, we have to define the helper class.
+                final byte[] classBinary = classToByteArray(helper);
+                return AccessController.doPrivileged(new PrivilegedAction<Class<?>>() {
+                    @Override
+                    public Class<?> run() {
+                        try {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Extract this nested try block into a separate method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1141)